larpandoracontent/LArMonitoring/EventClusterValidationAlgorithm.cc

```diff
@@ -0,0 +1,337 @@
+{
+    for (const CaloHit *const pCaloHit : caloHits)
+    {
+        const MCParticle *pMainMC {nullptr};
+        const MCParticleWeightMap &weightMap {pCaloHit->GetMCParticleWeightMap()};
+        float maxWeight {std::numeric_limits<float>::lowest()};
+        for (const auto &[pMC, weight] : weightMap)
+        {
+            if (weight > maxWeight)
+                pMainMC = pMC;
+        }
+        if (pMainMC)
+        {
+            hitParents[pCaloHit] = CaloHitParents();
+            hitParents[pCaloHit].m_pMainMC = pMainMC;
+        }
+    }
+
+    for (const Cluster *const pCluster : clusters)
+    {
+        const CaloHitList &isolatedHits {pCluster->GetIsolatedCaloHitList()};
+        CaloHitList clusterCaloHits;
+        pCluster->GetOrderedCaloHitList().FillCaloHitList(clusterCaloHits);
+        clusterCaloHits.insert(clusterCaloHits.end(), isolatedHits.begin(), isolatedHits.end());
+        for (const CaloHit *const pCaloHit : clusterCaloHits)
+        {
+            // Ignoring the calo hit if truth matching is missing
+            if (hitParents.find(pCaloHit) == hitParents.end())
+                continue;
+            hitParents[pCaloHit].m_pCluster = pCluster;
+        }
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+void EventClusterValidationAlgorithm::ApplyMCParticleMinSumHits(std::map<const CaloHit *const, CaloHitParents> &hitParents) const
+{
+    std::map<const MCParticle *const, int> mcSumHits;
+    for (const auto &[pCaloHit, parents] : hitParents)
+    {
+        const MCParticle *const pMainMC {parents.m_pMainMC};
+        if (mcSumHits.find(pMainMC) == mcSumHits.end())
+            mcSumHits[pMainMC] = 0;
+        mcSumHits.at(pMainMC)++;
+    }
+
+    for (auto it = hitParents.begin(), itNext = it; it != hitParents.end(); it = itNext)
+    {
+      itNext++;
+      if (mcSumHits.at(it->second.m_pMainMC) < m_minMCHitsPerView)
+        hitParents.erase(it);
+    }
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+std::map<const CaloHit *const, EventClusterValidationAlgorithm::CaloHitParents>
+EventClusterValidationAlgorithm::ApplyPDGCut(std::map<const CaloHit *const, CaloHitParents> &hitParents,
+                                             const ValidationType &valType) const
+{
+    std::map<const CaloHit *const, CaloHitParents> hitParentsValid;
+    for (const auto &[pCaloHit, parents] : hitParents)
+    {
+        const MCParticle *const pMainMC {parents.m_pMainMC};
+        if (valType != ValidationType::ALL)
+        {
+            const int pdg {std::abs(pMainMC->GetParticleId())};
+            if (
+                (valType == ValidationType::SHOWER && (pdg != 22 && pdg != 11)) ||
+                (valType == ValidationType::TRACK && (pdg == 22 || pdg == 11))
```

Comment:
As a note, among the Pandora SDK definitions (Pandora/PdgTable.h) are the defintions PHOTON and E_MINUS